src/main/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/AbstractOAuth2AuthenticationProvider.java

```diff
@@ -91,48 +95,80 @@ public String toString() {
-        OAuth20Service service = getService(state, redirectUrl);
+    /**
+     * Receive user data from OAuth2 provider after authn/z has been successfull. (Callback view uses this)
+     * Request a token and access the resource, parse output and return user details.
+     * @param code The authz code sent from the provider
+     * @param service The service object in use to communicate with the provider
+     * @return A user record containing all user details accessible for us
+     * @throws IOException Thrown when communication with the provider fails
+     * @throws OAuth2Exception Thrown when we cannot access the user details for some reason
+     * @throws InterruptedException Thrown when the requests thread is failing
+     * @throws ExecutionException Thrown when the requests thread is failing
+     */
+    public OAuth2UserRecord getUserRecord(String code, @NotNull OAuth20Service service)
+        throws IOException, OAuth2Exception, InterruptedException, ExecutionException {
+        
         OAuth2AccessToken accessToken = service.getAccessToken(code);
-
-        final String userEndpoint = getUserEndpoint(accessToken);
+    
+        if ( ! accessToken.getScope().contains(scope) ) {
+            // We did not get the permissions on the scope we need. Abort and inform the user.
+            throw new OAuth2Exception(200, BundleUtil.getStringFromBundle("auth.providers.orcid.insufficientScope"), "");
+        }
         
-        final OAuthRequest request = new OAuthRequest(Verb.GET, userEndpoint, service);
-        request.addHeader("Authorization", "Bearer " + accessToken.getAccessToken());
+        OAuthRequest request = new OAuthRequest(Verb.GET, getUserEndpoint(accessToken));
         request.setCharset("UTF-8");
-        
-        final Response response = request.send();
+        service.signRequest(accessToken, request);
+    
+        Response response = service.execute(request);
         int responseCode = response.getCode();
-        final String body = response.getBody();        
-        logger.log(Level.FINE, "In getUserRecord. Body: {0}", body);
-
-        if ( responseCode == 200 ) {
-            final ParsedUserResponse parsed = parseUserResponse(body);
-            return new OAuth2UserRecord(getId(), parsed.userIdInProvider,
-                                        parsed.username, 
-                                        OAuth2TokenData.from(accessToken),
-                                        parsed.displayInfo,
-                                        parsed.emails);
+        String body = response.getBody();
+        logger.log(Level.FINE, "In requestUserRecord. Body: {0}", body);
+        if ( responseCode == 200 && body != null ) {
+            return getUserRecord(body, accessToken, service);
         } else {
             throw new OAuth2Exception(responseCode, body, "Error getting the user info record.");
```

Comment:
Is this error user facing? Does it show up in the web interface? If so it should be moved into Properties.bundle at some point (though maybe not in this pull request if we would like to defer this refactoring).

Reply:
Should be fixed in c729a27